appengine/pond/duck/board/js/board.js

```diff
@@ -62,42 +74,56 @@ Pond.Duck.Board.createDuck = function() {
 };
 
 /**
- * Callback for when all the ducks for a user are fetched.s
+ * Callback for copy/delete action.
+ * @param {string} action Action attempted.
  */
-Pond.Duck.Board.getDucksCallback = function() {
-  var text;
-  if (this.status == 200) {
-    var duckList = JSON.parse(this.responseText);
-    Pond.Duck.Board.setTemplate(duckList)
-  } else {
-    text = BlocklyGames.getMsg('Games_httpRequestError') + '\nStatus: '
-        + this.status;
-  }
+Pond.Duck.Board.createGetDucksCallback = function(action) {
+  return function() {
+    var text;
+    if (this.status == 200) {
+      Pond.Duck.Board.refreshDuckList.call(this);
+      var meta = JSON.parse(this.responseText);
+      text = 'Duck ' + action + ' with key: ' + meta['duck_key'];
+    } else {
+      text = BlocklyGames.getMsg('Games_httpRequestError') + '\nStatus: '
+          + this.status;
+    }
+    document.getElementById('loading').style.display = 'none';
+    BlocklyDialogs.storageAlert(null, text);
+  };
 };
 
 /**
- * Set the template and add bind all events.
- * @param {Array<Object>} ducks A list holding all the ducks for
- * the current user. A duck consists of a name and a duckId.
+ * Callback function for AJAX call.
+ * Response contains a list holding all the ducks for the current user
+ * (name, duckId, and optionally ranking).
  */
-Pond.Duck.Board.setTemplate = function(ducks) {
-  document.body.innerHTML = Pond.Duck.Board.soy.start({}, null,
-    {
-      lang: BlocklyGames.LANG,
-      html: BlocklyGames.IS_HTML,
-      ducks: ducks
-    });
-  var copyBtns = document.getElementsByClassName('copyDuck');
-  var deleteBtns = document.getElementsByClassName('deleteDuck');
-  var createBtn = document.getElementById('createButton');
-  for (var i = 0; i < copyBtns.length; i++) {
-    BlocklyGames.bindClick(copyBtns[i], Pond.Duck.Board.copyDuck);
-  }
-  for (var i = 0; i < deleteBtns.length; i++) {
-    BlocklyGames.bindClick(deleteBtns[i], Pond.Duck.Board.deleteDuck);
+Pond.Duck.Board.refreshDuckList = function() {
+  // Convert the response data into something template will understand.
+  var responseData = JSON.parse(this.responseText);
+  var templateDucks = [];
+  for(var i = 0, duckData; (duckData = responseData["duckList"][i]); i++) {
+    var templateDuck = {
```

Comment:
I thought it was necessary, but it appears that I can instead pass the duckList as you suggest and use syntax: $duck['name'] instead of $duck.name in the template with no issues.